crates/uv-build-backend/src/lib.rs

```diff
@@ -229,17 +230,29 @@ fn find_roots(
 /// Returns the module root path, the directory below which the `__init__.py` lives.
 fn find_module_root(
     src_root: &Path,
-    module_name: Option<&Identifier>,
+    module_name: Option<&ModuleName>,
     package_name: &PackageName,
 ) -> Result<PathBuf, Error> {
-    let module_name = if let Some(module_name) = module_name {
+    let (module_name, stubs) = if let Some(module_name) = module_name {
         // This name can be uppercase.
-        module_name.to_string()
+        match module_name {
+            ModuleName::Identifier(module_name) => (module_name.to_string(), false),
+            ModuleName::Stubs(module_name) => (module_name.to_string(), true),
+        }
     } else {
-        // Should never error, the rules for package names (in dist-info formatting) are stricter
-        // than those for identifiers.
-        // This name is always lowercase.
-        Identifier::from_str(package_name.as_dist_info_name().as_ref())?.to_string()
+        // Infer stubs packages from package name alone. There are potential false positives if
+        // someone had a regular package with `-stubs`.
+        if let Some(stem) = package_name.to_string().strip_suffix("-stubs") {
```

Comment:
If a false positive happens, is there a way for a user to override that and say, "no, this is not a stubs package"?

Conversely, is there a way for a package that doesn't end with `-stubs` to be treated as a stubs package?

From this PR, it looks like the answer to both above is "no." Are we sure that's okay?

Reply:
In one direction, https://typing.python.org/en/latest/spec/distributing.html#stub-only-packages is a clear MUST about the `-stubs` suffix. In the other direction, I think I'd try this out, and if there are really packages with let's say `foo_stubs` we can add an override (`tool.uv.build-backend.stubs = False`). If we can have one option less by having this inference I'd prefer that.

Another option would be to require stubs packages to always declare a module name, so we can tell `-stubs` (stubs package) from `_stubs` (regular module).

Reply:
> In one direction, https://typing.python.org/en/latest/spec/distributing.html#stub-only-packages is a clear MUST about the `-stubs` suffix. In the other direction, I think I'd try this out, and if there are really packages with let's say `foo_stubs` we can add an override (`tool.uv.build-backend.stubs = False`). If we can have one option less by having this inference I'd prefer that.

I think that's good enough for me.

> Another option would be to require stubs packages to always declare a module name, so we can tell `-stubs` (stubs package) from `_stubs` (regular module).

I do like this idea! But I agree that inference probably makes more sense, and if necessary, we can add a way to override it if it comes up.

Reply:
I realized that we actually do have overrides: Setting the module name to `_stubs` vs. `-stubs` will trigger regular package vs. stubs package behavior.

Reply:
Nice. Yeah I think that SGTM!